zcash_primitives/src/transaction/mod.rs

```diff
@@ -24,7 +24,7 @@ const OVERWINTER_TX_VERSION: u32 = 3;
 const SAPLING_VERSION_GROUP_ID: u32 = 0x892F2085;
 const SAPLING_TX_VERSION: u32 = 4;
 
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[derive(Clone, Copy, Debug, PartialOrd, Ord, PartialEq, Eq, Hash)]
```

Comment:
A slightly unusual effect of this is that because transaction IDs are displayed byte-reversed, if a user sorts a `Vec<TxId>` and then displays each entry, the result is very much not sorted. Questions:

- Is this ideal behaviour?
- Does it matter to non-Display uses if we sorted in a way that ensured it was ordered for Display uses?
- Is it reasonable for `PartialOrd`'s behaviour to be altered later, if we decide we want to change the ordering?

Reply:
What I actually need is a way to de-dup `TxID`s. I thought it's easier to do it with `Vec::dedup` which seems efficient. 

The issue is that the protocol treats a `TxID` as a `uint256`, so it gets encoded as LittleEndian which reverses it, but it's really not an integer. For example, integer operations (like addition etc...) don't make sense for a TxID. 

Another thing that crossed my mind is that a `TxID` isn't really compatible with `Ord` in any meaningful sense. One `TxID` doesn't come "before" or "after" another in any meaningful way. 

Anyway, adding PartialOrd seemed like the easiest way to dedup them, but I'm open to suggestions